packages/text-field/src/TextField.js

```diff
@@ -137,11 +154,15 @@ export default class TextField extends Component {
               </label>
             )}
 
-            {this.props.showClearButton && (
-              <button
-                className="hig__text-field__clear-button hig__text-field--clear-button-visible"
-                onClick={this.props.onClearButtonClick}
-              />
+            {this.hasClearableInput() && (
```

Comment:
Instead of calling `hasClearableInput` twice, lets assign the result to a variable. The value shouldn't change within the render function.